src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -432,6 +437,41 @@ function isCustomComponent(tagName, props) {
   return tagName.indexOf('-') >= 0 || props.is != null;
 }
 
+/**
+ * Returns the children of this dom node in a way that is easily consumable for
+ * comparing to the component hierarchy.
+ * Unfortunately, a node's DOM children don't correspond exactly to its component's
+ * children. For example, text nodes in the component tree become 2 commment DOM
+ * nodes and an optional text dom node. Empty components become a single comment DOM node.
+ *
+ * Returns an array which contains domNodes and arrays of domNodes. Each item in
+ * the returned array represents either one or multiple dom nodes that should correspond
+ * to a single child node in the component tree.
+ */
+function getChildrenFromDomNode(parent) {
```

Comment:
I agree that the name isn't very descriptive or good.

I may be wrong, but it seems like `RenderedComponent` is generally a term in the code for actual components, and this method returns DOM nodes. What do you think about something like `getNativeNodeChildren` which mirrors the `getNativeNode` methods?